addon/-private/formatters/format-message.ts

```diff
@@ -51,65 +47,45 @@ function escapeOptions<T extends Record<string, unknown>>(object?: T) {
   return escapedOpts;
 }
 
+type MessageFormatValue = Parameters<IntlShape<string>['formatMessage']>[0]['defaultMessage'];
+type MessageFormatOptions = Parameters<IntlShape<string>['formatMessage']>[1];
+
 /**
  * @private
  * @hide
  */
-export default class FormatMessage {
+export default class FormatMessage extends Formatter<any> {
   static readonly type = 'message';
 
-  protected readonly config: FormatterConfig;
-  protected readonly readFormatConfig: () => Formats;
-
-  constructor(config: FormatterConfig) {
-    this.config = config;
-
-    // NOTE: a fn since we lazily grab the formatter from the config
-    // as it can change at runtime by calling intl.set('formats', {...});
-    this.readFormatConfig = config.readFormatConfig;
-  }
-
-  createNativeFormatter = memoize(
-    (ast: TranslationAST, locales: string | string[], formatConfig?: Partial<Formats>) => {
-      return new IntlMessageFormat(ast, locales, formatConfig, {
-        ignoreTag: true,
-      });
-    }
-  );
-
   // ! Function overloads are not passed through generic types for reasons that
   // evade my knowledge. ¯\_(ツ)_/¯
   // For this reason these types need to be manually copied over to the
   // `IntlService#formatMessage`.
   format(
     locale: string | string[],
-    maybeAst: string | TranslationAST,
-    options?: Partial<Record<string, unknown>> & { htmlSafe?: false }
+    maybeAst: MessageFormatValue,
+    options?: MessageFormatOptions & { htmlSafe?: false }
   ): string;
   format(
     locale: string | string[],
-    maybeAst: string | TranslationAST,
-    options: Partial<Record<string, unknown>> & { htmlSafe: true }
+    maybeAst: MessageFormatValue,
+    options: MessageFormatOptions & { htmlSafe: true }
   ): SafeString;
   format(
     locale: string | string[],
-    maybeAst: string | TranslationAST,
-    options?: Partial<Record<string, unknown>> & { htmlSafe?: boolean }
+    maybeAst: MessageFormatValue,
+    options?: MessageFormatOptions & { htmlSafe?: boolean }
   ): string | SafeString {
-    let ast = maybeAst as TranslationAST;
-
-    if (typeof maybeAst === 'string') {
-      // maybe memoize?  it's not a typical hot path since we
-      // parse when translations are pushed to ember-intl.
-      // This is only used if inlining a translation i.e.,
-      // {{format-message "Hi {name}"}}
-      ast = parse(maybeAst);
-    }
-
     const isHTMLSafe = options && options.htmlSafe;
-    const formatterInstance = this.createNativeFormatter(ast, locale, this.readFormatConfig());
+
     const escapedOptions = isHTMLSafe ? escapeOptions(options) : options;
-    const result = formatterInstance.format(escapedOptions) as string;
+    const result = this.getIntl(locale).formatMessage(
+      {
+        id: 'dummy',
```

Comment:
@formatjs/intl does translation lookup via ID but looks like the value here is already translated so dummy id is there so that it doesn't complain about missing ID.

I have to look at the translation store stuff to see how ember-intl handles its lookup and wire it in correctly.

Reply:
> I have to look at the translation store stuff to see how ember-intl handles its lookup and wire it in correctly.

Can think of it as `Map<locale: string, Map<key: string, translation: string>>`

Reply:
I think we still want to resolve this `dummy` id, should be able to thread the actual key (if there is one) back to format-message

Reply:
This is the case for inline `format-message 'a message'` instead of `t 'some.key'`. How does inline message get resolved to an ID?

Reply:
`t()` calls on this here: https://github.com/ember-intl/ember-intl/blob/master/addon/services/intl.js#L166

Was that your question?